src/main/java/reposense/util/StringsUtil.java

```diff
@@ -31,10 +31,33 @@ public static String replaceSpecialSymbols(String regexString, String replacemen
         return SPECIAL_SYMBOLS.matcher(regexString).replaceAll(replacementCharacter);
     }
 
-    public static String addQuote(String original) {
+    /**
+     * Adds quotes to the string.
+     */
+    public static String addQuotes(String original) {
         return "\"" + original + "\"";
     }
 
+    /**
+     * Adds the appropriate quotation marks for a file path depending on the OS.
+     */
+    public static String addQuotesForFilePath(String filePath) {
+        if (SystemUtil.isWindows()) {
+            return "\"" + filePath + "\"";
+        } else {
+            StringBuilder sb = new StringBuilder();
```

Comment:
Should we link https://stackoverflow.com/a/1250279 here as well?

Reply:
I think this is far more straightforward. It is a full single quote wrap while specially escaping single quotes. There are various ways to do it. Replacing `'` with `'\''` instead of `'"'"'` also works if I am not wrong. The test case is also written to catch this specific issue if someone tries to change it in the future.